eng/common/scripts/Verify-Links.ps1

```diff
@@ -257,21 +272,29 @@ while ($pageUrisToCheck.Count -ne 0)
 
   $linkUris = GetLinks $pageUri
   Write-Host "Found $($linkUris.Count) links on page $pageUri";
-  
+  $badLinksPerPage = @{};
   foreach ($linkUri in $linkUris) {
     $isLinkValid = CheckLink $linkUri
-    if (!$isLinkValid) {
-      $script:badLinks += $linkUri
+    if (!$isLinkValid -and !$badLinksPerPage.ContainsKey($linkUri)) {
+      $badLinksPerPage[$linkUri] = $true
     }
     if ($recursive -and $isLinkValid) {
       if ($linkUri.ToString().StartsWith($baseUrl) -and !$checkedPages.ContainsKey($linkUri)) {
         $pageUrisToCheck.Enqueue($linkUri);
       }
     }
   }
+  if ($badLinksPerPage.Count -gt 0) {
+    $badLinks[$pageUri] = $badLinksPerPage
+  }
 }
 
-Write-Host "Found $($checkedLinks.Count) links with $($badLinks.Count) broken"
-$badLinks | ForEach-Object { Write-Host "  $_" }
+Write-Host "Found $($checkedLinks.Count) links with $($badLinks.Count) pages broken."
```

Comment:
If $badLinks.Count -gt 0 we should change this summary to be a LogError so it shows up in red.